src/state/tls.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""gateway-api-integrator resource definition."""
+
+import dataclasses
+
+import ops
+from ops.jujuversion import JujuVersion
+from ops.model import Relation
+
+TLS_CERTIFICATES_INTEGRATION = "certificates"
+
+
+class TlsIntegrationMissingError(Exception):
+    """Exception raised when _situation_.
+
+    Attrs:
+        msg (str): Explanation of the error.
+    """
+
+    def __init__(self, msg: str):
+        """Initialize a new instance of the TlsIntegrationMissingError exception.
+
+        Args:
+            msg (str): Explanation of the error.
+        """
+        self.msg = msg
+
+
+@dataclasses.dataclass(frozen=True)
+class TLSInformation:
+    """A component of charm state containing resource definition for kubernetes secret.
+
+    Attrs:
+        tls_requirer_integration: The integration instance with a TLS provider.
+        tls_certs: A dict of hostname: certificate obtained from the relation.
+        tls_keys: A dict of hostname: private_key stored in juju secrets.
+    """
+
+    tls_requirer_integration: Relation
+    tls_certs: dict[str, str]
+    tls_keys: dict[str, str]
+
+    @classmethod
+    def from_charm(cls, charm: ops.CharmBase) -> "TLSInformation":
+        """Create a resource definition from charm instance.
```

Comment:
I think this line is not correct as you are not creating a resource definition here.

Reply:
Fixed the docstring. Thanks!